File: lib/src/main/java/me/ameriod/lib/mvp/view/deligate/ViewMvpDelegate.java
package me.ameriod.lib.mvp.view.deligate;

import android.os.Parcelable;

import me.ameriod.lib.mvp.Mvp;

public interface ViewMvpDelegate<V extends Mvp.View, P extends Mvp.Presenter<V>> {

    void onAttachedToWindow();

    void onDetachedFromWindow();

    void onRestoreInstanceState(Parcelable state);

    Parcelable onSaveInstanceState();

    P getPresenter();
}

File: lib/src/main/java/me/ameriod/lib/mvp/deligate/ViewMvpDelegate.kt
package me.ameriod.lib.mvp.deligate

import android.os.Parcelable

import me.ameriod.lib.mvp.Mvp

interface ViewMvpDelegate<in V : Mvp.View, out P : Mvp.Presenter<in V>> {

    fun onAttachedToWindow()

    fun onDetachedFromWindow()

    fun onRestoreInstanceState(state: Parcelable)

    fun onSaveInstanceState(): Parcelable

    fun getPresenter(): P
}
